From abd09a616fa7a0ce9000ced3d33d1ea83567e86e Mon Sep 17 00:00:00 2001
From: Laurent Fasnacht <fasnacht@protonmail.ch>
Date: Mon, 10 Feb 2020 10:17:27 +0000
Subject: [PATCH] scanner: fix indesc_list stack to be in the correct order

This fixes the location displayed in error messages.

Signed-off-by: Laurent Fasnacht <fasnacht@protonmail.ch>
Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
(cherry picked from commit 10aaa1130c2a574c8eebb0593651a9ee54db1021)
Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/scanner.l | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/scanner.l b/src/scanner.l
index dc20cd3d79d43..6852c19c4179b 100644
--- a/src/scanner.l
+++ b/src/scanner.l
@@ -667,7 +667,11 @@ addrstring	({macaddr}|{ip4addr}|{ip6addr})
 static void scanner_push_indesc(struct parser_state *state,
 				struct input_descriptor *indesc)
 {
-	list_add_tail(&indesc->list, &state->indesc_list);
+	if (!state->indesc)
+		list_add_tail(&indesc->list, &state->indesc_list);
+	else
+		list_add(&indesc->list, &state->indesc->list);
+
 	state->indesc = indesc;
 	state->indesc_idx++;
 }
-- 
2.28.0

